src/Pgvector/Vector.cs

```diff
@@ -4,22 +4,60 @@
 
 namespace Pgvector;
 
-public class Vector
+public class Vector : IEquatable<Vector>
 {
-    private ReadOnlyMemory<float> vec;
+    public ReadOnlyMemory<float> Memory { get; }
```

Comment:
This exposes the ReadOnlyMemory publicly, so that users (as well as the plugin's converter) can access the data without copying (which is what ToArray does below)